js/globals.ts

```diff
@@ -55,8 +55,12 @@ window.setInterval = timers.setInterval;
 // being used, which it cannot statically determine within this module.
 window.Blob = blob.DenoBlob;
 export type Blob = blob.DenoBlob;
-window.File = file.DenoFile;
-export type File = file.DenoFile;
+
+// TODO(ry) Do not export a class implementing the DOM, export the DOM
```

Comment:
You should be able to do something like this, following the pattern below:

https://github.com/denoland/deno/blob/e1faccc83e5e00a97bad082df21a499cda2d927c/js/globals.ts#L72-L73

You will need a `FileConstructor` interface in `domTypes` though.

Reply:
I will leave this for a follow up PR as it's out of scope WRT removing extension support. I've added a link to your comment in the source.